src/llfs/storage_file_builder.test.cpp

```diff
@@ -56,6 +56,114 @@ class StorageFileBuilderTest : public ::testing::Test
+      i64 expected_page0_offset =
+          llfs::PackedConfigBlock::kPayloadCapacity + sizeof(llfs::PackedConfigBlock::crc64);
+
+      // Iterate through each PackedPageDeviceConfig
+      //
+      for (const llfs::PackedConfigSlot& slot : p_config_block.slots) {
+        // Verify PackedPageDeviceConfig
+        //
+        ASSERT_EQ(slot.tag, llfs::PackedConfigSlotBase::Tag::kPageDevice);
+        const llfs::PackedPageDeviceConfig& page_device_config =
+            llfs::config_slot_cast<llfs::PackedPageDeviceConfig>(slot);
+        ASSERT_EQ(page_device_config.page_0_offset, expected_page0_offset);
+        ASSERT_EQ(page_device_config.device_id, expected_device_id);
+        ASSERT_EQ(page_device_config.page_count, (little_i64)kTestPageCount);
+        ASSERT_EQ(page_device_config.page_size_log2, page_size_log2);
+
+        // Update page offset numbers and device ID
+        //
+        expected_page0_offset +=
+            page_device_config.page_size() * kTestPageCount - llfs::PackedConfigSlot::kSize;
+        expected_device_id += 1;
+      }
+
+      // Assuming 1 slot filled if back slot, all slots filled otherwise.
+      //
+      constexpr u64 kMaxPageDeviceSlots =
+          llfs::PackedConfigBlock::kPayloadCapacity / llfs::PackedPageDeviceConfig::kSize;
+      u64 expected_num_slots = (config_block == config_blocks.back()) ? 1 : kMaxPageDeviceSlots;
+
+      // Set offsets of config block based on position
+      //
+      if (config_block == config_blocks.front()) {
+        expected_prev_offset = llfs::PackedConfigBlock::kNullFileOffset;
+      } else {
+        expected_prev_offset = -expected_next_offset;
+      }
+      if (config_block == config_blocks.back()) {
+        expected_next_offset = llfs::PackedConfigBlock::kNullFileOffset;
+      } else {
+        expected_next_offset = batt::round_up_bits(
+            12, expected_page0_offset + llfs::PackedPageDeviceConfig::kSize * expected_num_slots);
+      }
+
+      // Verify the packed config block
+      //
+      ASSERT_EQ(p_config_block.magic, llfs::PackedConfigBlock::kMagic);
+      ASSERT_EQ(p_config_block.version, llfs::make_version_u64(0, 1, 0));
+      ASSERT_EQ(p_config_block.prev_offset, expected_prev_offset);
+      ASSERT_EQ(p_config_block.next_offset, expected_next_offset);
+      ASSERT_EQ(p_config_block.slots.size(), expected_num_slots);
+      ASSERT_EQ(p_config_block.crc64, p_config_block.true_crc64());
+    }
+  }
+
+  // Writes random data to disk given the file, offset, and amount of data to write. Used to
+  // randomize data of a test file to verify repeated write and read testing.
+  //
+  void write_rand_data(llfs::IoRingRawBlockFile& test_file, i64 write_offset, i64 write_size)
+  {
+    constexpr i64 kBufferSize = 32768;
+    i64 write_bytes_remaining = batt::round_up_bits(9, write_size);
+    i64 offset = batt::round_up_bits(llfs::PackedConfigBlock::kSize, write_offset);
```

Comment:
first arg to `round_up_bits` should be the log2(kSize), not kSize.

Reply:
No longer rounding up in function, added parameter check of input and note to the function doc.